src/main/java/be/syntra/devshop/DevshopFront/services/CartServiceImpl.java

```diff
@@ -48,53 +57,69 @@ public CartDto getCart() {
     }
 
     @Override
-    public void addToCart(Product product) {
-        product.setTotalInCart(product.getTotalInCart() + 1);
-        List<Product> productList = getCart().getProducts();
-        productList.add(product);
-        getCart().setProducts(productList);
+    public void addToCart(Long id) {
+        List<CartProductDto> contentDtoList = getCart().getCartProductDtoList();
```

Comment:
IF your contentDtoList would be a Set. Which I recommend you refactor, because we don't want our Cart to have the ability to contain duplicates... then your whole addToCart method could look like this: 
CartProductDto currentCartProductDto = contentDtoList.stream()
                    .filter(cartProductDto -> cartProductDto.getProductDto().getId().equals(id))
                    .findFirst()
                    .orElse(CartProductDto.builder()
                            .productDto(productMapper.convertToProductDto(productService.findById(id)))
                            .count(0)
                            .build());
            currentCartProductDto.setCount(currentCartProductDto.getCount() + 1);
            contentDtoList.add(currentCartProductDto);
And the productNotInCart method could be removed as well

Reply:
changed List into Set in both CartDto and CartDisplayDto, fixed a bunch of tests, changed a bunch of thymeleaf things too for that.   Implemented your code :-)
Also changed the BE to sets of products
And to think that there was a moment that i tought : i need to change this list into a set :-|